File: 06. Sorting/Insertion_Sort.py
def insertion_sort(arr):
    for i in range(1, len(arr)):
        # Set key:
        key = arr[i]

        j = i - 1
        while j >= 0 and arr[j] > key:
            # Swap:
            arr[j + 1] = arr[j]
            arr[j] = key
            
            # Decrement 'j':
            j -= 1

    return arr


array = [5, 2, 12, 12, 1]
print("Original array: %s" % array)
print("Sorted array: %s" % insertion_sort(array))

File: 06. Sorting/InsertionSort.go
package main

import "fmt"

func main() {
   
    arr := []int{3, -2, 2, 6, 5, 4, -1, 7, -3}

	fmt.Println("Before sorting")
	for _, val := range(arr) {
		fmt.Printf("%v ", val)
	}

    insertionSort(arr)
}

func insertionSort(arr []int) {
    len := len(arr)
    for i := 1; i < len; i++ {
        for j := 0; j < i; j++ {
            if arr[j] > arr[i] {
                arr[j], arr[i] = arr[i], arr[j]
            }
        }
    }
    
    fmt.Printf("\nAfter Sorting\n")
    for _, val := range arr {
        fmt.Printf("%v ", val)
    }
}
